src/libelektra/keyhelpers.c

```diff
@@ -345,108 +345,113 @@ int keyInit(Key *key)
  * initialized, but really empty, object
  * @param va the variadic argument list
  */
-void keyVInit (Key *key, const char *name, va_list va)
+void keyVInit(Key *key, const char *name, va_list va)
 {
 	keyswitch_t action = 0;
+	size_t value_size = 0;
 	void *value = 0;
-	int valueSizeChanged = 0;
-	size_t valueSize = 0;
+	void (*func) (void) = 0;
+	uint64_t flags = 0;
 	char *owner = 0;
-	option_t nameOptions = 0;
-	void (*p) (void) = 0;
 
-	if (!key) return;
+	if (!key)
+		return;
 
 	if (name) {
-		action=va_arg(va, keyswitch_t);
-		while (action) {
+		while ((action = va_arg(va, keyswitch_t))) {
 			switch (action) {
+				/* flags with an argument */
 				case KEY_SIZE:
-					valueSize=va_arg(va, size_t);
-					valueSizeChanged = 1;
-					break;
-				case KEY_BINARY:
-					keySetMeta (key, "binary", "");
+					value_size = va_arg(va, size_t);
 					break;
 				case KEY_VALUE:
 					value = va_arg(va, void *);
-					if (valueSizeChanged && keyIsBinary(key))
-					{
-						keySetBinary(key,value, valueSize);
-					} else if (keyIsBinary(key)) {
-						valueSize = elektraStrLen (value);
-						keySetBinary(key,value, valueSize);
-					} else {
-						keySetString(key,value);
-					}
+					if (value_size)
+						keySetBinary(key, value, value_size);
+					else if (keyIsBinary(key))
+						keySetBinary(key, value, elektraStrLen(value));
+					else
+						keySetString(key, value);
 					break;
-#ifndef WIN32
-				case KEY_UID:
-					keySetUID(key,va_arg(va,uid_t));
+				case KEY_FUNC:
+					func = va_arg(va, void(*)(void));
+					keySetBinary(key, &func, sizeof(func));
 					break;
-				case KEY_GID:
-					keySetGID(key,va_arg(va,gid_t));
+				case KEY_META:
+					value = va_arg (va, char *);
+					/* First parameter is name */
+					keySetMeta(key, value, va_arg(va, char *));
 					break;
-				case KEY_MODE:
-					/* Theoretically this should be mode_t, but prefer using
-					   int to avoid troubles when sizeof(mode_t)!=sizeof(int)
-					 */
-					keySetMode(key,va_arg(va, int));
+
+				/* flags without an argument */
+				case KEY_FLAGS:
+					flags = va_arg(va, uint64_t);
```

Comment:
Questionable, this would prevent overriding. Since this currently works: keyNew(..., KEY_NAME, "user(foo", KEY_NAME, "user/bar"), I decided to make KEY_FLAGS behave the same.